src/test/java/hudson/tasks/junit/HistoryTest.java

```diff
@@ -108,7 +108,87 @@ public void testFailedSince() throws Exception {
         PackageResult smallPackage = tr5.byPackage("org.jvnet.hudson.examples.small");
         ClassResult miscClass = smallPackage.getClassResult("MiscTest");
         CaseResult eleanorCase = miscClass.getCaseResult("testEleanor");
+        assertNotNull(eleanorCase);
         assertTrue("eleanor failed", !eleanorCase.isPassed());
-        assertEquals("eleanor has failed since build 3", 3, eleanorCase.getFailedSince()); 
+        assertEquals("eleanor has failed since build 3", 3, eleanorCase.getFailedSince());
+    }
+
+    @LocalData
+    @Test
+    public void testSkippedSince() throws Exception {
+        assertNotNull("project should exist", project);
```

Comment:
It is quite hard to understand if the setup is not visible and packed into a zip file. (But it is not your fault, as this approach has been used in the existing test case as well.)

Reply:
Completely agree :) it took running the test and examining its log in order to understand that a zip file was used..